### PR TITLE
📦 NEW: Support For Multiple Cities in Pakistan

### PR DESCRIPTION
**Pull request by mian-muhammad**
Added support for almost any city of Pakistan. Instead of importing data from json files, scrapping it from urdupoint's website.

### CODE REVIEW

**Comment by mian-muhammad** [comment]
Sure, will look into that

**Comment by mian-muhammad** [comment]
@ahmadawais changes made. Scrapped data and updated the data directory.

**Comment by mian-muhammad** [comment]
I am looking into Geo's website, there few things I want to say:
1. There is no consistency in h2's
2. They are only providing data for 23 cities

Where as on [Urdupoint](https://www.urdupoint.com/islam/pakistan-ramadan-calendar-sehar-aftar-timings.html) there data is quite consistent for scraping and they providing data for around 200 cities which includes KPK's cities. Also in KPK's few cities first Ramadan was on 24th April. This might be the reason few data files were of 24th insted of 25th April.

image

**Comment by mian-muhammad** [comment]
@ahmadawais also if you allow me I can post scraper for generating data but that is made on [urdupoint's website](https://www.urdupoint.com/islam/pakistan-ramadan-calendar-sehar-aftar-timings.html)

**Comment by ahmadawais** [comment]
I don't want the GEO site. Your data is wrong. I don't care what source you use as far as the data is correct. I mentioned Geo as a reference. Please read all of my comment again. Add the script that scrapes and remove the data files. Also the scrapper should produce wrote data otherwise the roza number would be wrong.

**Comment by mian-muhammad** [comment]
oh my bad, actually there 29 entities for Ramadan calendar on the source I was using, that's why scraper was creating 29 days instead of 30.

**Comment by ahmadawais** [comment]
Then find a way to fix it, use another source, find an API if you can. But in the current form, the data that gets generated is incomplete and wrong. Which makes it difficult for me to accept this PR despite all the time and effort you put into this.

**Comment by mian-muhammad** [comment]
I am looking into that @ahmadawais

File: package.json [hunk 1]
```diff
@@ -33,10 +33,13 @@
 	},
 	"scripts": {
 		"test": "ava",
-		"format": "prettier --write \"./**/*.{js,json}\""
+		"format": "prettier --write \"./**/*.{js,json}\"",
+		"get-data": "node ./utils/generateData.js"
 	},
 	"dependencies": {
+		"axios": "^0.19.2",
```
Comment (ahmadawais) [inline package.json:40]: This is a `devDependency`

File: package.json [hunk 2]
```diff
@@ -33,10 +33,13 @@
...
 		"chalk": "^4.0.0",
+		"cheerio": "^1.0.0-rc.3",
```
Comment (ahmadawais) [inline package.json:42]: This is a `devDependency`

File: readme.md [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 - 🌜 Get Sehar/Iftar times for your city
 - 🗓 Get Sehar/Iftar times for entire month of Ramadan
-- 🇵🇰 10+ major cities data including `lahore`, `multan`, `karachi`, `islamabad`, `rawalpindi`, `quetta`, `peshawar`, `hyderabad`,`faisalabad`, `gujranwala`, `sialkot`, `ghotki`
+- 🇵🇰 99 cities data including `lahore`, `multan`, `karachi`, `islamabad`, `rawalpindi`, `quetta`, `peshawar`, `hyderabad`,`faisalabad`, `gujranwala`, `sialkot`, `ghotki`
```
Comment (ahmadawais) [inline readme.md:14]:
```suggestion
- 🇵🇰 99 cities data including `lahore`, `multan`, `karachi`, `islamabad`, `rawalpindi`, `quetta`, `peshawar`, `hyderabad`,`faisalabad`, `gujranwala`, `sialkot`, `ghotki`, and many more
```

**Comment by mian-muhammad** [comment]
@ahmadawais 

- Data files removed

- Added scraper which scrapes data from [hmariweb](https://hamariweb.com/islam/pakistan_ramadan-timings1.aspx?Page=1), for 99 cities.

- Added an npm script `npm run get-data` which will generate data files for 99 cities replacing the old ones and will also update `cities.js` file.

- Made minor changes in `readme.md`

- Support for multi word cities

File: index.js [hunk 4]
```diff
@@ -17,11 +17,13 @@ const notFound = `${sym.error} ${input}: not found.\n Consider contributing!`;
 	init();
 	input === 'help' && (await cli.showHelp(0));
 
-	const city = input ? input.toLowerCase() : `lahore`;
+	const city = input ? input.toLowerCase().replace(/\s+/g, '-') : `lahore`;
+	const cityNiceName = `${city.charAt(0).toUpperCase()}${city.slice(1).replace(/-/g, ' ')}`;
+	console.log(cityNiceName);
```
Comment (ahmadawais) [inline index.js:22]: Don't log anything.

File: utils/theEnd.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
 const sym = require('log-symbols');
-const {dim} = require('chalk');
+const { dim } = require('chalk');
 
-module.exports = async ({city, noData}) => {
-	const cityNiceName = `${city.charAt(0).toUpperCase()}${city.slice(1)}`;
+module.exports = async ({ city, noData }) => {
+	const cityNiceName = `${city.charAt(0).toUpperCase()}${city.slice(1).replace(/-/g, ' ')}`;
```
Comment (ahmadawais) [inline utils/theEnd.js:5]: This should have the `cityNiceName` otherwise where is it used?

**Comment by ahmadawais** [comment]
@mian-muhammad the problem I mentioned is the same. You didn't address it. I'm currently checking the data for Lahore and it's wrong. The first ramadan is shown as 24th April and 30th of Ramadan is shown as 23rd May — this is all wrong. 

When I run `npx ramadan-cli` it shows me the wrong data of 12th May instead of 13th May that would be today because you are adding an extra roza in the start which is wrong.

The idea is great. I love the work you have done here. But I can't use it if the data produced by this work is wrong. Can you fix this issue for all cities and request a review once that's done?

**Comment by mian-muhammad** [comment]
@ahmadawais sorry I was not aware of the behavior of toISOString, that's why dates were messed up. I have updated the code as you asked:

- Added `cheerios` and `axios` as `devDependencies`

- Removed additional log statements 

- Fixed the date issue

- Update `readme.md` file

**Comment by ahmadawais** [comment]
@mian-muhammad I'm sorry I was not able to entertain your PR. Did you end up fixing the data?

**Pull request closed without merging**